### PR TITLE
Clearified keycloak steps:

### PR DESCRIPTION
**Pull request by coltoneshaw**
#### Summary
Worked with a few customers on setting up Keycloak. Ran into some problems, so this is just an attempt to clarify that.

https://community.mattermost.com/boards/team/rcgiyftm7jyrxnma1osd8zswby/bbusacfknofye5jgtmrufo75mnc/v5hakifb3i7nw5emhjtbrdua7uy/c3kcnho4k3p877pd9tz9fh6bpmo

### CODE REVIEW

File: source/onboard/sso-saml-keycloak.rst [hunk 1]
```diff
@@ -13,6 +13,11 @@ The following process provides steps to configure SAML with Keycloak for Matterm
 Set up a connection app for Mattermost SSO
 -------------------------------------------
 
+.. note:: 
+    This was tested with Keycloak v18.0.0.
+
+    Additionally, you do not typically have to make a realm dedicated to Mattermost. We recommend adding Mattermost as a client to your primary realm.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:19]:
```suggestion
    Additionally, you don't typically need to make a realm dedicated to Mattermost. We recommend adding Mattermost as a client to your primary realm.
```

File: source/onboard/sso-saml-keycloak.rst [hunk 2]
```diff
@@ -13,6 +13,11 @@ The following process provides steps to configure SAML with Keycloak for Matterm
...
+
 1. Log in to Keycloak as an administrator.
 
 2. Click **Clients** then **Create** and save. You'll use this client ID in a later step.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:23]:
```suggestion
2. Select **Clients**, then **Create** and **Save**. You'll use this client ID in a later step.
```

File: source/onboard/sso-saml-keycloak.rst [hunk 3]
```diff
@@ -38,13 +43,17 @@ Set up a connection app for Mattermost SSO
 
 6. Select **Generate new keys**.
 
-7. Select **Export**, use the values below, and then select **Download**.
+7. Select **Export**, use the values below, and then select **Download**. Once you've downloaded it navigate back to the Mattermost Client in Keycloak for the next steps.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:46]:
```suggestion
7. Select **Export**, use the values below, and then select **Download**. Once you've downloaded it, navigate back to the Mattermost client in Keycloak for the next steps. 
```

File: source/onboard/sso-saml-keycloak.rst [hunk 4]
```diff
@@ -38,13 +43,17 @@ Set up a connection app for Mattermost SSO
...
-7. Select **Export**, use the values below, and then select **Download**.
+7. Select **Export**, use the values below, and then select **Download**. Once you've downloaded it navigate back to the Mattermost Client in Keycloak for the next steps. 
 
     - **Archive Format**: ``PKCS12``
     - **Key Alias**: ``mattermost``
     - **Key Password**: ``mattermost``
     - **Store Password**: ``mattermost``
 
+    .. note::
+        In the below picture we used `Mattermost` for the `Realm Certificate Alias`. You can use any value that you would like here, as it's designed to identify what this certificate is within your Keycloak realm.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:54]:
```suggestion
        In the image below, we used `Mattermost` for the `Realm Certificate Alias`. You can use any value that you would like here, as it's designed to identify what this certificate is within your Keycloak realm.
```

File: source/onboard/sso-saml-keycloak.rst [hunk 5]
```diff
@@ -60,6 +69,8 @@ Set up a connection app for Mattermost SSO
 
 9. Add the username and ID attribute.
 
+    If you're planning to sync your SAML users with LDAP within Mattermost the ID value used here must match with your Mattermost LDAP ``ID Attribute``.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:72]:
```suggestion
    If you're planning to sync your SAML users with LDAP within Mattermost, the ID value used here must match with your Mattermost LDAP ``ID Attribute``.
```

File: source/onboard/sso-saml-keycloak.rst [hunk 6]
```diff
@@ -71,6 +82,10 @@ Set up a connection app for Mattermost SSO
 .. image:: ../../source/images/keycloak_4_create_username_attribute.png
    :alt: In Keycloak, on the Mappers tab, create a protocol mapper, then save your changes.
 
+    Once done your Mappers should look like this:
+.. image:: ../../source/images/keycloak_4_create_username_attribute_finished.png
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:86]:
```suggestion
.. image:: ../../source/images/keycloak_4_create_username_attribute_finished.png
   :alt: Example of protocol mapper configuration. 
```

File: source/onboard/sso-saml-keycloak.rst [hunk 7]
```diff
@@ -125,7 +140,9 @@ Configure SAML for Mattermost
 .. image:: ../../source/images/keycloak_7_mattermost_request_signing.png
    :alt: In the System Console, you can optionally request signing with configured parameters.
 
-6. Set attributes for the SAML Assertions, which will update user information in Mattermost. Attributes for email and username are required to match the values you configured in Keycloak in steps 9 and 10. See `documentation on SAML configuration settings </configure/configuration-settings.html#saml>`__ for more detail.
+6. Set attributes for the SAML Assertions, which will update user information in Mattermost. 
+    
+    The below are from step 8 and 9 above. These values must be the **SAML Attribute Name** within Keycloak. See `documentation on SAML configuration settings </configure/configuration-settings.html#saml>`__ for more detail.
```
Comment (cwarnermm) [inline source/onboard/sso-saml-keycloak.rst:145]:
```suggestion
    The attributes below are from steps 8 and 9 above. These values must be the **SAML Attribute Name** within Keycloak. See `documentation on SAML configuration settings </configure/configuration-settings.html#saml>`__ for more details.
```

**Review by cwarnermm — COMMENTED** [review]
Thanks, @coltoneshaw! I've provided editorial suggestions inline.

**Comment by coltoneshaw** [comment]
@cwarnermm - I think I broke a record on number of changes requested right away! I committed all of them. Let me know if you see anything else.

**Review by cwarnermm — APPROVED** [review]

**Comment by cwarnermm** [comment]
@coltoneshaw - I need your help with one small fix. The new screenshot needs to be renamed from ``source/images/keycloak_4_create_username_attribute_finished.png.jpg`` to match the reference in the body of the docs page.

**Comment by coltoneshaw** [comment]
@cwarnermm - Done. I left a  `.png.jpg` for some reason!

**Pull request merged**